roles/contiv_network/tasks/main.yml

```diff
@@ -1,44 +1,13 @@
 ---
 # This role contains tasks for configuring and starting netmaster and netplugin service
+- name: check dns container image
+  shell: docker inspect skynetservices/skydns
+  register: docker_inspect_result
+  ignore_errors: yes
 
-- name: download netmaster and netplugin
-  get_url:
-    validate_certs: no
-    url: https://github.com/contiv/netplugin/releases/download/v0.0.0-12-11-2015.20-54-40.UTC/netplugin-v0.0.0-12-11-2015.20-54-40.UTC.tar.bz2
-    dest: /tmp/contivnet.tar.bz2
+- name: pull dns container image
+  shell: docker pull skynetservices/skydns
+  when: "'No such image' in docker_inspect_result.stderr"
 
-- name: install netmaster and netplugin
-  shell: tar vxjf /tmp/contivnet.tar.bz2
-  args:
-    chdir: /usr/bin/
-    creates: netmaster
-
-- name: copy environment file for netplugin
-  template: src=netplugin.j2 dest=/etc/default/netplugin
-
-- name: copy systemd units for netplugin
-  copy: src=netplugin.service dest=/etc/systemd/system/netplugin.service
-
-- name: start netplugin
-  shell: systemctl daemon-reload && systemctl start netplugin
-
-- name: setup netmaster host alias
-  shell: echo "{{ service_vip }} netmaster" >> /etc/hosts
-
-- name: copy environment file for netmaster
-  copy: src=netmaster dest=/etc/default/netmaster
-  when: run_as == "master"
-
-- name: copy systemd units for netmaster
-  copy: src=netmaster.service dest=/etc/systemd/system/netmaster.service
-  when: run_as == "master"
-
-# resolves an issue with netmaster: https://github.com/contiv/netplugin/issues/210
-- name: start netmaster
-  shell: systemctl daemon-reload && systemctl start netmaster
-  when: run_as == "master"
-  run_once: true
-
-# FIXME: need to move the following to correct roles -- also needs release tooling
-- name: install contivctl
-  shell: source /etc/profile.d/00golang.sh && go get github.com/contiv/contivctl
+- include: deploy_binaries.yml
+  when: contiv_demo 
```

Comment:
@DivyaVavili  as we have discussed in slack I don't think we should make make deployment conditional in ansible. It should be done in the project as deploying a dev environment should be limited to the project.

Let's chat on slack, I can help to figure that out.